src/main/java/com/gooddata/warehouse/WarehouseS3Credentials.java

```diff
@@ -62,52 +58,32 @@
     public WarehouseS3Credentials(final String region,
                                   final String accessKey,
                                   final String secretKey) {
-        this(region, accessKey, secretKey, null, null, null);
-    }
-
-    /**
-     * Used to list saved S3 credentials (not intended for the end user)
-     *
-     * @param region    S3 region
-     * @param accessKey S3 access key
-     * @param updatedBy URI of the user who updated the credentials last
-     * @param updated   last modification date
-     */
-    public WarehouseS3Credentials(final String region,
-                                  final String accessKey,
-                                  final String updatedBy,
-                                  final DateTime updated) {
-        this(region, accessKey, null, updatedBy, updated, null);
+        this(region, accessKey, secretKey, null, null);
     }
 
     @JsonCreator
-    WarehouseS3Credentials(@JsonProperty("region") final String region,
-                           @JsonProperty("accessKey") final String accessKey,
-                           @JsonProperty("secretKey") final String secretKey,
-                           @JsonProperty("updatedBy") final String updatedBy,
-                           @JsonProperty("updated") @JsonDeserialize(using = ISODateTimeDeserializer.class) final DateTime updated,
-                           @JsonProperty("links") final Map<String, String> links) {
+    protected WarehouseS3Credentials(@JsonProperty("region") final String region,
+                                     @JsonProperty("accessKey") final String accessKey,
+                                     @JsonProperty("secretKey") final String secretKey,
+                                     @JsonProperty("updated") @JsonDeserialize(using = ISODateTimeDeserializer.class)
+                                         final DateTime updated,
+                                     @JsonProperty("links") final Map<String, String> links) {
```

Comment:
I suggest to make the links a typesfe structure, not a generic map.

Reply:
done